src/firebase-resource-manager.ts

```diff
@@ -0,0 +1,178 @@
+import * as clc from "cli-color";
+
+import * as api from "./api";
+import * as FirebaseError from "./error";
+import * as logger from "./logger";
+import { LroPoller, LroPollerOptions } from "./lro-poller";
+import { OraWrapper } from "./oraWrapper";
+
+const ONE_SECOND_MILLIS = 1000;
+
+export enum ParentResourceType {
+  ORGANIZATION = "organization",
+  FOLDER = "folder",
+}
+
+export interface ParentResource {
+  id: string;
+  type: ParentResourceType;
+}
+
+export class FirebaseResourceManager {
+  private poller: LroPoller = new LroPoller();
+
+  async createFirebaseProject(
+    projectId: string,
+    projectDisplayName: string,
+    parentResource?: ParentResource
+  ): Promise<{ projectNumber: string; projectId: string }> {
+    const cloudProjectCreationOperation = await this.createCloudProject(
```

Comment:
nit: The wrapping can be reduced to two lines:

```ts
const cloudProjectCreationOperation =
  await this.createCloudProject(projectId, projectDisplayName, parentResource);
```

Reply:
Your suggestion is complained by the `inter`

Reply:
Okay, I suppose we'll stick with the linter.